### PR TITLE
Version display respect formatter settings

### LINKED ISSUE #85: Display format of Version does not respect width/precision settings.
The following should all pass, but currently just all formatted to the same string `"1.0.0-rc1"`.

``` rust
let version = Version::parse("1.0.0-rc1").unwrap();
assert_eq!(format!("{:20}", version), "1.0.0-rc1           ");
assert_eq!(format!("{:*^20}", version), "*****1.0.0-rc1******");
assert_eq!(format!("{:.4}", version),  "1.0.");
```

### PR DESCRIPTION
**Pull request by opilar**
Should resolve #85

### CODE REVIEW

File: src/version.rs [hunk 1]
```diff
@@ -565,6 +567,14 @@ mod tests {
     }
 
     #[test]
+    fn test_display() {
+        let version = Version::parse("1.2.3-rc1").unwrap();
+        assert_eq!(format!("{:20}", version), "1.2.3-rc1           ");
```
Comment (opilar) [inline src/version.rs]: The test fails here. I don't understand why.

Comment (KodrAus) [reply src/version.rs]: What's the actual output you're getting here?

Comment (opilar) [reply src/version.rs]:
```
---- version::tests::test_display stdout ----
	thread 'version::tests::test_display' panicked at 'assertion failed: `(left == right)`
  left: `"1.2.3-rc1"`,
 right: `"1.2.3-rc1           "`', src/version.rs:572:8
```

Comment (KodrAus) [reply src/version.rs]: Oh right. So the format isn't respecting the padding length... I quickly tried to run the code today but Windows said no to compiling `libgit-sys`.

**Comment by steveklabnik** [comment]
Hi @opilar , do you intend to get this passing? It'd be great to have, but if you don't have the time, I understand!

**Comment by opilar** [comment]
@steveklabnik still stuck in the same place. I intend to resole it, but at this weekend.
Can you help me to get why does the test fail?

**Comment by opilar** [comment]
@steveklabnik Fixed now. Can you review it please?

**Pull request merged**